pkgsMusl.glib fails to build

### Steps To Reproduce
Steps to reproduce the behavior:
1. build `pkgsMusl.glib` (I used x86_64-linux)

### Build log

(Just the relevant part)

```
FAILED: glib/tests/cxx-98.p/cxx.cpp.o
g++ -Iglib/tests/cxx-98.p -Iglib/tests -I../glib/tests -I. -I.. -Iglib -I../glib -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -Wduplicated-branches -Wimplicit-fallthrough -Wmisleading-indentation -Wunused -Wno-unused-parameter -Wno-cast-function-type -Wno-pedantic -Wno-format-zero-length -Wno-variadic-macros -Werror=format=2 -Werror=init-self -Werror=missing-include-dirs -Werror=pointer-arith -Werror=unused-result -pthread '-DG_LOG_DOMAIN="GLib"' -UG_DISABLE_ASSERT -std=c++98 -MD -MQ glib/tests/cxx-98.p/cxx.cpp.o -MF glib/tests/cxx-98.p/cxx.cpp.o.d -o glib/tests/cxx-98.p/cxx.cpp.o -c ../glib/tests/cxx.cpp
../glib/tests/cxx.cpp: In function ‘int main(int, char**)’:
../glib/tests/cxx.cpp:201:15: error: missing sentinel in function call [-Werror=format=]
  201 |   g_test_init (&argc, &argv, NULL);
      |   ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
cc1plus: some warnings being treated as errors
[196/1304] Compiling C object glib/libglib-2.0.so.0.7400.1.p/gvariant.c.o
[197/1304] Compiling C++ object glib/tests/cxx.p/cxx.cpp.o
[198/1304] Compiling C++ object glib/tests/cxx-03.p/cxx.cpp.o
FAILED: glib/tests/cxx-03.p/cxx.cpp.o
g++ -Iglib/tests/cxx-03.p -Iglib/tests -I../glib/tests -I. -I.. -Iglib -I../glib -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -Wduplicated-branches -Wimplicit-fallthrough -Wmisleading-indentation -Wunused -Wno-unused-parameter -Wno-cast-function-type -Wno-pedantic -Wno-format-zero-length -Wno-variadic-macros -Werror=format=2 -Werror=init-self -Werror=missing-include-dirs -Werror=pointer-arith -Werror=unused-result -pthread '-DG_LOG_DOMAIN="GLib"' -UG_DISABLE_ASSERT -std=c++03 -MD -MQ glib/tests/cxx-03.p/cxx.cpp.o -MF glib/tests/cxx-03.p/cxx.cpp.o.d -o glib/tests/cxx-03.p/cxx.cpp.o -c ../glib/tests/cxx.cpp
../glib/tests/cxx.cpp: In function ‘int main(int, char**)’:
../glib/tests/cxx.cpp:201:15: error: missing sentinel in function call [-Werror=format=]
  201 |   g_test_init (&argc, &argv, NULL);
      |   ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
cc1plus: some warnings being treated as errors

```

We should check in the upstream glib repo whether this has already been reported and fixed, and if not, submit our own fix upstream and then apply it in Nixpkgs.

--- a/pkgs/development/libraries/glib/default.nix
+++ b/pkgs/development/libraries/glib/default.nix
@@ -68,6 +68,19 @@ stdenv.mkDerivation (finalAttrs: {
   ] ++ lib.optionals stdenv.hostPlatform.isMusl [
     ./quark_init_on_demand.patch
     ./gobject_init_on_demand.patch
+
+    # Fix error about missing sentinel in glib/tests/cxx.cpp
+    # These two commits are part of already merged glib MRs 3033 and 3031:
+    # https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3033
+    (fetchpatch {
+      url = "https://gitlab.gnome.org/GNOME/glib/-/commit/0ca5254c5d92aec675b76b4bfa72a6885cde6066.patch";
+      sha256 = "OfD5zO/7JIgOMLc0FAgHV9smWugFJuVPHCn9jTsMQJg=";
+    })
+    # https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3031
+    (fetchpatch {
+      url = "https://gitlab.gnome.org/GNOME/glib/-/commit/7dc19632f3115e3f517c6bc80436fe72c1dcdeb4.patch";
+      sha256 = "v28Yk+R0kN9ssIcvJudRZ4vi30rzQEE8Lsd1kWp5hbM=";
+    })
   ] ++ [
     ./glib-appinfo-watch.patch
     ./schema-override-variable.patch

